From: Johannes Stezenbach <js@sig21.net>
Date: Wed Jan 19 04:43:34 EST 2011
Subject: [PATCH RFC] rt2800: disable powersaving as default

Powersaving is broken on rt2800usb (the hardware cannot do timed wakeup
to receive beacons) and also on rt2800pci currently (auto-wakeup irq
does no seem to work, for reasons so far unknown).

Signed-off-by: Johannes Stezenbach <js at sig21.net>

diff --git a/drivers/net/wireless/rt2x00/rt2800lib.c b/drivers/net/wireless/rt2x00/rt2800lib.c
index b7d91d5..b846fd9 100644
--- a/drivers/net/wireless/rt2x00/rt2800lib.c
+++ b/drivers/net/wireless/rt2x00/rt2800lib.c
@@ -3229,10 +3229,9 @@ int rt2800_probe_hw_mode(struct rt2x00_dev *rt2x00dev)
 	u16 eeprom;
 
 	/*
-	 * Disable powersaving as default on PCI devices.
+	 * Disable powersaving as default.
 	 */
-	if (rt2x00_is_pci(rt2x00dev) || rt2x00_is_soc(rt2x00dev))
-		rt2x00dev->hw->wiphy->flags &= ~WIPHY_FLAG_PS_ON_BY_DEFAULT;
+	rt2x00dev->hw->wiphy->flags &= ~WIPHY_FLAG_PS_ON_BY_DEFAULT;
 
 	/*
 	 * Initialize all hw fields.
